service/frontend/workflow_handler.go

```diff
@@ -4273,30 +4284,42 @@ func (wh *WorkflowHandler) getHistoryReverse(
 	nextPageToken []byte,
 	branchToken []byte,
 ) (*historypb.History, []byte, int64, error) {
+
+	// readFullPageEventsReverse reads a full page of history events from history service in reverse orcer. Due to
+	// storage format of V2 History it is not guaranteed that pageSize amount of data is returned. Function returns the
+	// list of history events, the size of data read, the next page token, and an error if present.
+	readFullPageEventsReverse := func(
+		req *historyservice.ReadHistoryBranchReverseRequest,
+	) ([]*historypb.HistoryEvent, int, []byte, error) {
+		var historyEvents []*historypb.HistoryEvent
+		size := 0
+		for {
+			response, err := wh.historyClient.ReadHistoryBranchReverse(ctx, req)
+			if err != nil {
+				return nil, 0, nil, err
+			}
+			historyEvents = append(historyEvents, response.HistoryEvents...)
+			size += int(response.Size_)
```

Comment:
Why do we need to return this? Can't the caller just use `len(historyEvents)`?

Reply:
=) copied from `history_manager_util.go` verbatim